src/Numerics/Mesh/FTP.jl

```diff
@@ -0,0 +1,201 @@
+using DocStringExtensions
+using KernelAbstractions
+using CUDA
+
+include("FTP_kernels.jl")
+
+"""
+    ftpxv!(grd::DiscontinuousSpectralElementGrid,
+        mesh::Symbol,
+        opts::Symbol,
+        tropts::Bool,
+        vin::AbstractArray{FT},
+        vout::AbstractArray{FT},
+        vin_den::Union{AbstractArray{FT}, Nothing} = nothing,
+    ) where {FT<:AbstractFloat}
+
```

Comment:
Can we get this down to a function of the form
```
ftpxv!(vout, vin, (A, B, C), temp)
```
that computes `vout[rst,...] = vin[ijk,...] * A[r,i] * B[s,j] * C[t,k]`?

Reply:
I was thinking the same thing, and wondering if we could use [Tullio.jl](https://github.com/mcabbott/Tullio.jl) (or if we could open a PR in Tullio).